protocols/abci/addressing-1174.md

```diff
@@ -0,0 +1,241 @@
+leverage that extensive body of research applied to the Tendermint algorithm.
+This situation is risky and undesirable.
+
+So, the questions that arise are the following.
+Can we come up with a set of weaker requirements
+that applications unable to fulfill the current ABCI 1.0 requirements
+can still fulfill?
+Can we maintain modularity with these new requirements?
+Is this set of weaker requirements still strong enough to guarantee consensus _termination_?
+
+## Solution Proposed
+
+### Modified consensus _validity_
+
+Function $valid(v, bc_{h-1})$, as explained above, exclusively depends on its inputs
+(a block, and the blockchain state at the previous height).
+So it is always supposed to provide the same result when called at the same height for the same inputs,
+no matter at which process.
+This was the main reason for introducing the determinism requirements on `ProcessProposal`
+in the ABCI 1.0 specification.
+
+If we are to relax the determinism requirements on the application,
+we first need to modify function $valid()$ to be of the form $valid(v, bc_{h-1}, x_p)$,
+where $x_p$ is a variable local to process $p$.
+As $x_p$ may be different from $x_q$ for two processes $p$ and $q$, using
+$valid(v, bc_{h-1}, x_p)$ to implement in `PrepareProposal`  and `ProcessProposal`
+may break Requirements 3 to 5.
+
+Consensus _validity_ property is then modified as follows:
+
+- _weak validity_: function $valid(v, bc_{h-1}, x_p)$ has returned _true_ at least once
+  by a correct process for the decided block $v$.
+
+### Eventual Requirements
+
+We now relax the relevant ABCI 1.0 requirements in the following way.
+
+* Requirement 3b [`PrepareProposal`, `ProcessProposal`, eventual coherence]:
+  There exists a time $ts_{h}$ for every height $h$ such that,
+  for any two correct processes $p$ and $q$ and any round $r_p$ in height $h$ starting after $ts_{h}$,
+  if $q$'s CometBFT calls `RequestProcessProposal` on $u_p$,
+  $q$'s application returns _Accept_ in `ResponseProcessProposal`.
+
+* The determinism-related requirements, namely requirements 4 and 5, are removed.
+
+We call $ts_{h}$ the coherence-stabilization time for height $h$.
+
+If we think in terms of $valid(v, bc_{h-1}, x_p)$, notice that it is the application's responsibility
+to ensure 3b, that is, the application designers need to prove that the $x_p$ values at correct processes
+are evolving in a way that eventually `ResponseProcessProposal` returns _Accept_ at all correct processes
+that call `RequestPropocessPropsal` for height $h$ after $ts_{h}$.
+For instance, in Proposer-Based Timestamp, $x_p$ can be considered to be process $p$'s local clock,
+and having clocks synchronized is the mechanism ensuring eventual acceptance of a proposal.
+Finally, it is worth noting that _weak validity_ requires just one correct processes while requirement 3b
+refers to all correct processes.
+
+> [TODO: can we _tighten up_ eventual coherence?... i.e., make it even weaker?]
```

Comment:
I can hardly imagine a weaker form for it. We are already defining a valid value as a value that has been found valid by a single correct process. In the implementation we are stronger, as we require 2f+1, namely f+1 correct.

Reply:
The only way in which I can think is to make the common acceptance happen infinitely often rather than in every round after $ts_h$.